plugins/modules/pnp_workflow_manager.py

```diff
@@ -1259,6 +1189,189 @@ def verify_diff_deleted(self, config):
         self.status = "success"
         return self
 
+    def get_device_list_pnp(self, serial_number):
+        """
+        Get the PNP device list from the Cisco Catalyst Center.
+
+        Parameters:
+          - self (object): An instance of the class containing the method.
+          - serial_number (str): Serial number contains string from the device data.
+        Returns:
+          The method returns an instance of the class with updated attributes:
+          - self.device_response: Gets the device_id and stores it
+
+        Example:
+          passing device details and getting pnp device details response
+        """
+        try:
+            response = self.dnac_apply['exec'](
+                family="device_onboarding_pnp",
+                function='get_device_list',
+                params={"serial_number": serial_number},
+                op_modifies=True,
+            )
+
+            if response and isinstance(response, list) and len(response) == 1:
+                self.device_response = response[0]
+                self.log("Successfully retrieved PNP device details for serial number: {0}".format(serial_number), "INFO")
+                return self.device_response
+
+            msg = "No device found with serial number: {0}".format(serial_number)
+            self.log(msg, "WARNING")
+            self.module.fail_json(msg=msg)
+
+        except Exception as e:
+            msg = "An error occurred while retrieving device with serial number {0}: {1}".format(serial_number, str(e))
+            self.log(msg + str(e), "WARNING")
+            self.module.fail_json(msg=msg)
+
+    def get_device_by_id_pnp(self, device_id):
+        """
+        Get the PNP device details using by device id from the Cisco Catalyst Center.
+
+        Parameters:
+        - self (object): An instance of the class containing the method.
+        - device_id (str): Device id contains id from the device data
+        Returns:
+        The method returns an instance of the class with updated attributes:
+        - self.device_response: Gets the device_id and stores it
+
+        Example:
+        passing device details and getting pnp device details response
+        """
+        self.log("Attempting to retrieve PNP device details for device id: {0}".format(device_id), "INFO")
+        try:
+            device_details_response = self.dnac_apply['exec'](
+                family="device_onboarding_pnp",
+                function="get_device_by_id",
+                params={"id": device_id},
+                op_modifies=True,
```

Comment:
Same here.. To get the device list we don't need to set to True

Reply:
The op_modifies parameter is typically used to indicate whether the operation being performed is expected to modify the state of the system or the device in some way. In the context of a get function, which usually retrieves information, setting op_modifies to True might not be necessary or even appropriate, since get operations are generally read-only and do not change any data.

Reply:
removed the line